tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -2640,6 +2641,111 @@ def test_get_dataset_id_version_4xx(self):
 
 
+class TestGetDatasetManifest(BaseAPIPortalTest):
+    def test_get_manifest_ok(self):
+        collection = self.generate_published_collection()
+        collection_id = collection.collection_id
+        published_revision = self.generate_revision(collection_id)
+
+        published_dataset_revision = self.generate_dataset(
+            collection_version=published_revision,
+            publish=True,
+            artifacts=[
+                DatasetArtifactUpdate(DatasetArtifactType.H5AD, "http://mock.uri/asset.h5ad"),
+                DatasetArtifactUpdate(DatasetArtifactType.ATAC_FRAGMENT, "http://mock.uri/atac_frags.bgz"),
+            ],
+        )
+
+        test_url = f"/curation/v1/collections/{published_dataset_revision.collection_id}/datasets/{published_dataset_revision.dataset_id}/manifest"
+        response = self.app.get(test_url)
+
+        self.assertEqual(200, response.status_code)
+
+        assert response.json == {
+            "anndata": f"http://domain/{published_dataset_revision.dataset_version_id}.h5ad",
+            "atac_fragment": f"http://domain/{published_dataset_revision.dataset_version_id}.bgz",
+        }
+
+    def test_get_manifest_cases_ok(self):
+        cases = [
+            {
+                "artifacts": {
+                    "atac_fragment": DatasetArtifactUpdate(
+                        DatasetArtifactType.ATAC_FRAGMENT, "http://mock.uri/atac_frags.bgz"
+                    )
+                },
+                "name": "fragments_only",
+            },
+            {
+                "artifacts": {"anndata": DatasetArtifactUpdate(DatasetArtifactType.H5AD, "http://mock.uri/asset.h5ad")},
+                "name": "anndata_only",
+            },
+            {
+                "artifacts": {
+                    "anndata": DatasetArtifactUpdate(DatasetArtifactType.H5AD, "http://mock.uri/asset.h5ad"),
+                    "atac_fragment": DatasetArtifactUpdate(
+                        DatasetArtifactType.ATAC_FRAGMENT, "http://mock.uri/atac_frags.bgz"
+                    ),
+                },
+                "name": "anndata_and_fragments",
+            },
+            {"artifacts": {}, "name": "no_artifacts"},
+        ]
+        for case in cases:
+            with self.subTest(f"Get manifest case: {case['name']}"):
+
+                collection = self.generate_unpublished_collection()
+
+                dataset = self.generate_dataset(
+                    collection_version=collection,
+                    artifacts=list(case["artifacts"].values()),
+                )
+
+                test_url = f"/curation/v1/collections/{dataset.collection_id}/datasets/{dataset.dataset_id}/manifest"
+                response = self.app.get(test_url)
+                self.assertEqual(200, response.status_code)
+
+                expected = {
+                    k: f"http://domain/{dataset.dataset_version_id}.{v.type}" for k, v in case["artifacts"].items()
```

Comment:
Since we are using the artifact_id for the fragment file, this test will need to change to assert the fragments is named with the artifact_id.